File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/inject/intro/V8Engine.kt
package io.micronaut.docs.inject.intro

import javax.inject.Singleton

// tag::class[]
@Singleton// <2>
class V8Engine : Engine {

    override var cylinders = 8
    override fun start(): String {
        return "Starting V8"
    }
}
// end::class[]

File: test-suite-python/src/test/python/micronaut/docs/inject/typed/V8Engine.py

from jakarta.inject import Singleton
from micronaut.context.annotation import Bean
from .Engine import Engine

# tag::class[]
@Singleton # <2>
@Bean(typed = Engine)
class V8Engine(Engine):
    cylinders: int = 8

    def start(self) -> str:
        return "Starting V8"

    def get_cylinders(self) -> int:
        return self.cylinders
# end::class[]
